core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2621,7 +2621,14 @@ public Expression implement(RexToLixTranslator translator, RexCall call,
             operand0.getType().getSqlTypeName() == SqlTypeName.TIMESTAMP
                 ? BuiltInMethod.ADD_MONTHS
                 : BuiltInMethod.ADD_MONTHS_INT;
-        return Expressions.call(method.method, trop0, trop1);
+        if (typeName == SqlTypeName.TIME) {
+          return Expressions.convert_(
+                  Expressions.subtract(trop0,
+                    Expressions.constant(0)),
+                    (Class) long.class);
```

Comment:
cast is not needed.

Reply:
Thank you for your review. I will revise it

Reply:
hi @hsyuan I have submitted PR again, please review it again
Thanks
qianjin

Reply:
thanks, LGTM

Reply:
@hsyuan thanks